ocaml-lsp-server/src/semantic_highlighting.ml

```diff
@@ -10,142 +10,80 @@ open Fiber.O
    - [ ] support textDocument/semanticTokens/range *)
 
 module Token_type : sig
-  type t = private int
-
-  val module_ : t
-
-  val type_ : t
+  type t
 
-  val class_ : t
+  val of_builtin : SemanticTokenTypes.t -> t
 
-  val enum : t
+  val module_ : t
 
   val module_type : t
 
-  val record : t
-
-  val type_parameter : t
-
-  val parameter : t
-
-  val variable : t
-
-  val property : t
-
-  val enum_member : t
-
-  val event : t
-
-  val function_ : t
-
-  val method_ : t
-
-  val macro : t
-
-  val keyword : t
-
-  val modifier : t
-
-  val comment : t
-
-  val string : t
-
-  val number : t
-
-  val regexp : t
-
-  val operator : t
-
-  val decorator : t
-
-  val list : string list
+  val to_int : t -> int
 
   val to_legend : t -> string
-end [@warning "-32"] = struct
-  type t = int
-
-  let module_ = 0
```

Comment:
I think we could also just make this code more self-evident by replacing a magic int with something like `List.findi legend "namespace"` and not bother with having `of_builtin` everywhere

You may also be bothered by large modules - I guess this could be better just moved into a separate file

Reply:
> You may also be bothered by large modules - I guess this could be better just moved into a separate file

Yes, I'm more bothered by the amount of code this adds. I'd rather us focus towards adding a proper semantic tokens api to the `lsp` module and just using whatever it provides. Once we do that, we can focus on optimizing and improving the API.

Reply:
`lsp` could use extensible variants nicely for this, I think